nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ForkRecord.java

```diff
@@ -0,0 +1,293 @@
+        relationships.add(REL_ORIGINAL);
+        relationships.add(REL_FAILURE);
+        relationships.add(REL_FORK);
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String resultPathText = context.getProperty(RECORD_PATH).evaluateAttributeExpressions(flowFile).getValue();
+        final RecordPath resultRecordPath = recordPathCache.getCompiled(resultPathText);
+
+        final RecordReaderFactory readerFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+        final RecordSetWriterFactory writerFactory = context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+        final boolean addParentFields = context.getProperty(INCLUDE_PARENT_FIELDS).asBoolean();
+
+        final FlowFile original = flowFile;
+        final FlowFile outFlowFile = session.create(original);
+        final AtomicInteger readCount = new AtomicInteger(0);
+        final AtomicInteger writeCount = new AtomicInteger(0);
+
+        try {
+
+            session.read(flowFile, new InputStreamCallback() {
+                @Override
+                public void process(final InputStream in) throws IOException {
+                    try (final RecordReader reader = readerFactory.createRecordReader(original, in, getLogger())) {
+
+                        final RecordSchema writeSchema = writerFactory.getSchema(original, reader.getSchema());
+                        final OutputStream out = session.write(outFlowFile);
+
+                        try (final RecordSetWriter recordSetWriter = writerFactory.createWriter(getLogger(), writeSchema, original, out)) {
+
+                            recordSetWriter.beginRecordSet();
+
+                            // we read each record of the input flow file
+                            Record record;
+                            while ((record = reader.nextRecord()) != null) {
+
+                                readCount.incrementAndGet();
+
+                                // evaluate record path in each record of the flow file
+                                Iterator<FieldValue> it = resultRecordPath.evaluate(record).getSelectedFields().iterator();
+
+                                while(it.hasNext()) {
+                                    FieldValue fieldValue = it.next();
+                                    RecordFieldType fieldType = fieldValue.getField().getDataType().getFieldType();
+
+                                    // we want to have an array here, nothing else allowed
+                                    if(fieldType != RecordFieldType.ARRAY) {
+                                        getLogger().debug("The record path " + resultPathText + " is matching a field "
```

Comment:
In this case, it seems to me like this indicates an improper configuration of the processor and should probably go to failure, rather than silently skipping the record. At least, I would want to make it optional. Thoughts?